causal_testing/testing/causal_test_suite.py

```diff
@@ -0,0 +1,27 @@
+from collections import UserDict
+
+
+class CausalTestSuite(UserDict):
+    """
+    A CausalTestSuite is an extension of the UserDict class, therefore it behaves as a python dictionary with the added
+    functionality of this class.
+    The dictionary structure should be the keys are base_test_cases representing the treatment and outcome Variables,
+    and the values are test objects. Test Objects hold a causal_test_case_list which is a list of causal_test_cases
+    which provide control and treatment values, and an iterator of Estimator Class References
+
+    This dictionary can be fed into the CausalTestEngines execute_test_suite function which will iterate over all the
+    base_test_case's and execute each causal_test_case with each iterator.
+    """
+
+    def add_test_object(self, base_test_case, causal_test_case_list, estimators, estimate_type: str = "ate"):
```

Comment:
It would be nice to have that resilience for non intentional stuff. I'm not sure how a multi-key `__setitem__` would work for a dictionary subclass without explicitly calling the `__setitem__` method however?

Reply:
Yeah - you're right. Forget this suggestion!